src/grt/src/GlobalRouter.cpp

```diff
@@ -221,17 +221,25 @@ void GlobalRouter::saveCongestion()
   }
   std::ofstream out(congestion_file_name_);
 
-  std::vector<std::pair<GSegment, TileCongestion>> congestionGridsV,
-      congestionGridsH;
-  fastroute_->getCongestionGrid(congestionGridsV, congestionGridsH);
+  std::vector<CongestionInformation> congestionGridsV, congestionGridsH;
+  auto nets_in_congested_edges = fastroute_->findCongestedEdgesNets();
+  fastroute_->getCongestionGrid(
+      congestionGridsV, congestionGridsH, nets_in_congested_edges);
   for (auto& it : congestionGridsH) {
+    const auto& [seg, tile, srcs] = it;
     out << "violation type: Horizontal congestion\n";
-    const int capacity = it.second.first;
-    const int usage = it.second.second;
-    out << "\tsrcs: \n";
+    const int capacity = tile.first;
+    const int usage = tile.second;
+    if (!srcs.empty()) {
+      out << "\tsrcs: ";
+      for (const auto& net : srcs) {
```

Comment:
I've decided to just use a `std::set` instead of `std::unordered_set`, since we're only adding values there once. I think it should not affect the runtime, and it prevents the necessity of copying the unordered set content to a new struct for sorting.